### PR TITLE
Extract codeObjects into a World object

### PR DESCRIPTION
**Pull request by raphaelpierquin**

### CODE REVIEW

File: world.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    var playground = {
+      name: name,
+      codeObject: function(name,initCode) {
+        if (codeObjects[name]) return codeObjects[name];
+        var code = initCode ? initCode : "";
+        var codeObject = {
+          name: name,
+          playground: playground,
+          code: function() { return code; },
+          updateCode: function(updatedCode, silently) {
```
Comment (jonathanperret) [inline world.js]: Also, if we're going to have a setter, why not call it `setCode`?

File: world.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            if (! silently) notifyUpdate(this);
+          },
+          delete: function(silently) {
+            delete codeObjects[name];
+            if (playground.isEmpty())
+              delete playgrounds[playground.name];
+            if (!silently) notifyDelete(this);
+          },
+        };
+        codeObjects[name]=(codeObject);
```
Comment (jonathanperret) [inline world.js]: `()`

File: world.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    notifyUpdate = fn;
+  }
+
+  function onCodeObjectDelete(fn) {
+    notifyDelete = fn;
+  }
+
+  function playground(name)  {
+    if (playgrounds[name]) return playgrounds[name];
+    var codeObjects = {};
```
Comment (jonathanperret) [inline world.js]: `Object.create(null)`

File: world.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+module.exports = function() {
+
+  var playgrounds = {},
```
Comment (jonathanperret) [inline world.js]: `Object.create(null)`

File: world.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      notifyUpdate = function(){},
+      notifyDelete = function(){}
+
+  function tour() {
+    return Object.keys(playgrounds);
+  }
+
+  function onCodeObjectUpdate(fn) {
```
Comment (jonathanperret) [inline world.js]: Also, I understand why observability (`onUpdate`/`onDelete`) may be desirable in the near future, but it is dead code for now! Maybe try to use it in `app.js` to notify clients?

**Comment by raphaelpierquin** [comment]
refactored a big part of the initial code. Please review.

**Comment by raphaelpierquin** [comment]
Thankyou for the review !
I made a big refactoring as the result of your review. Will create a new PR with a single commit.

**Pull request closed without merging**